server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/rfc/model/Id.scala

```diff
@@ -0,0 +1,32 @@
+/** **************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ * *
+ * http://www.apache.org/licenses/LICENSE-2.0                 *
+ * *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ * ***************************************************************/
+package org.apache.james.jmap.rfc.model
+
+import eu.timepit.refined.api.Refined
+import eu.timepit.refined.boolean.And
+import eu.timepit.refined.collection.Size
+import eu.timepit.refined.numeric.Interval
+import eu.timepit.refined.string.MatchesRegex
+import be.venneborg.refined.play.RefinedJsonFormats._
+
+object Id {
+  type Id = String Refined And[
+    Size[Interval.Closed[1, 255]],
+    MatchesRegex["^[a-zA-Z0-9-_]*$"]]
+}
```

Comment:
Don't we miss a test class for this?

Reply:
Actually it tested in Duc Tran PR, but for now maybe are two different versions.May be i will add test for this, WDYT?

Reply:
Yes please, and test the null case please. As said here: https://github.com/linagora/james-project/pull/3215#issuecomment-396290997
you might need to use `Option` for it

Reply:
No don't care about the null, I think Remi said enough it would fail with a JSerror when you deserialize in that case (right?) just care about other cases (I think Duc Tran's WIP on Refined could help?)